src/ts_catalog/continuous_agg.c

```diff
@@ -504,8 +504,10 @@ continuous_agg_fill_bucket_function(int32 mat_hypertable_id, ContinuousAggsBucke
 
 		/* Bucket function */
 		Assert(!isnull[AttrNumberGetAttrOffset(Anum_continuous_aggs_bucket_function_function)]);
-		bf->bucket_function = DatumGetObjectId(
+		const char *bucket_function_str = TextDatumGetCString(
 			values[AttrNumberGetAttrOffset(Anum_continuous_aggs_bucket_function_function)]);
+		bf->bucket_function = DatumGetObjectId(
+			DirectFunctionCall1(regprocedurein, CStringGetDatum(bucket_function_str)));
```

Comment:
Since this is written in the metadata, it can be wrong, or become wrong during an upgrade, so it is probably better to generate an error if the lookup fails rather than continuing with an invalid Oid.

Reply:
Make sense, will raise an error in case of an InvalidOid be returned... just to clarify that I've started to work on a following PR to get rid of this metadata table and instead read all the time bucket information from the QueryTree.

Reply:
Actually `regprocedurein` already check if the function exists and if not it raise an exception. Added a test for it: https://github.com/timescale/timescaledb/pull/6940/files#diff-2abad157593301155140079119454d6a436a528a616af5607cccfe1c379c8ad0R558-R567